nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -863,257 +1119,11 @@ impl DataStoreInventoryTest for DataStore {
-                use db::schema::sw_caboose::dsl;
-                dsl::sw_caboose
-                    .filter(dsl::id.eq_any(sw_caboose_ids))
-                    .limit(sql_limit)
-                    .select(SwCaboose::as_select())
-                    .load_async(&**conn)
-                    .await
-                    .context("loading sw_cabooses")?
-                    .into_iter()
-                    .map(|sw_caboose_row| {
-                        (
-                            sw_caboose_row.id,
-                            Arc::new(nexus_types::inventory::Caboose::from(
-                                sw_caboose_row,
-                            )),
-                        )
-                    })
-                    .collect()
-            };
-            limit_reached =
-                limit_reached || cabooses_by_id.len() == usize_limit;
-
-            // Assemble the lists of cabooses found.
-            let mut cabooses_found = BTreeMap::new();
-            for c in inv_caboose_rows {
-                let by_baseboard = cabooses_found
-                    .entry(nexus_types::inventory::CabooseWhich::from(c.which))
-                    .or_insert_with(BTreeMap::new);
-                let Some(bb) = baseboards_by_id.get(&c.hw_baseboard_id) else {
-                    bail!(
-                        "unknown baseboard found in inv_caboose: {}",
-                        c.hw_baseboard_id
-                    );
-                };
-                let Some(sw_caboose) = cabooses_by_id.get(&c.sw_caboose_id)
-                else {
-                    bail!(
-                        "unknown caboose found in inv_caboose: {}",
-                        c.sw_caboose_id
-                    );
-                };
-
-                let previous = by_baseboard.insert(
-                    bb.clone(),
-                    nexus_types::inventory::CabooseFound {
-                        time_collected: c.time_collected,
-                        source: c.source,
-                        caboose: sw_caboose.clone(),
-                    },
-                );
-                anyhow::ensure!(
-                    previous.is_none(),
-                    "duplicate caboose found: {:?} baseboard {:?}",
-                    c.which,
-                    c.hw_baseboard_id
-                );
-            }
-
-            Ok((
-                Collection {
-                    id,
-                    errors,
-                    time_started,
-                    time_done,
-                    collector,
-                    baseboards: baseboards_by_id.values().cloned().collect(),
-                    cabooses: cabooses_by_id.values().cloned().collect(),
-                    sps,
-                    rots,
-                    cabooses_found,
-                },
-                limit_reached,
-            ))
-        }
-        .boxed()
-    }
 }
 
 #[cfg(test)]
 mod test {
+    use crate::context::OpContext;
```

Comment:
What do you think about adding a test that uses read_all_or_nothing with too-short a limit to make sure it fails in the expected way?  (I'm hopeful this could be like 5 lines in the existing test.)

Reply:
Done in [8ad101f](https://github.com/oxidecomputer/omicron/pull/4504/commits/8ad101fff837f1373947e10b207367b34bf7932f)